pkg/lib/cockpit.js

```diff
@@ -2718,16 +2714,28 @@ function factory() {
      * Dark mode
      */
     (function() {
-        if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
-            cockpit.dark_mode = true;
-        } else {
-            cockpit.dark_mode = false;
-        }
+        const theme = localStorage.getItem('shell:theme') || 'auto';
+        setTheme(theme);
+
+        window.onstorage = event => {
+            if (event.key !== "shell:theme")
+                return;
+
+            setTheme(event.newValue);
+        };
 
         window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', event => {
+            const theme = localStorage.getItem('shell:theme') || 'auto';
+            // If a user set's an explicit theme, ignore system changes.
+            if (theme !== "auto")
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17914-20221114-163017-11a54652-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)